builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -84,6 +86,7 @@ type Provisioner struct {
 	ServerURL            string      `mapstructure:"server_url"`
 	SkipInstall          bool        `mapstructure:"skip_install"`
 	SSLVerifyMode        string      `mapstructure:"ssl_verify_mode"`
+	EnableReporting      string      `mapstructure:"enable_reporting"`
```

Comment:
Wouldn't it make more sense to define this like:

```go
DisableReporting bool `mapstructure:"disable_reporting"`
```

This way it's really clear that by default this is on (which is the default for Chef as well), and at the same time we can make better use of Go's type safety.

Of course the docs should then also be updated just a little and the template should be changed to something like:

```go
{{ if .DisableReporting }}enable_reporting false{{ end }}
```